src/gif/AnimatedGIFSource.ts

```diff
@@ -0,0 +1,215 @@
+import { decompressFrames, type ParsedFrame, parseGIF } from 'gifuct-js';
+import { DOMAdapter } from '../environment/adapter';
+import { CanvasSource } from '../rendering/renderers/shared/texture/sources/CanvasSource';
+import { Texture } from '../rendering/renderers/shared/texture/Texture';
+
+/**
+ * Represents a single frame of a GIF. Includes image and timing data.
+ * @memberof gif
+ */
+interface AnimatedGIFFrame
+{
+    /** Image data for the current frame */
+    texture: Texture<CanvasSource>;
+    /** The start of the current frame, in milliseconds */
+    start: number;
+    /** The end of the current frame, in milliseconds */
+    end: number;
+}
+
+/**
+ * Options when constructing from buffer
+ * @memberof gif
+ */
+interface AnimatedGIFBufferOptions
+{
+    /** FPS to use when the GIF animation doesn't define any delay between frames */
+    fps: number;
+}
+
+/**
+ * Resource provided to AnimatedGIF instances. This is very similar to using a shared
+ * Texture between Sprites. This source contains all the frames and animation needed
+ * to support playback.
+ * @memberof gif
+ */
+class AnimatedGIFSource
+{
+    /** Width of the animation */
+    public readonly width: number;
+
+    /** Height of the animation */
+    public readonly height: number;
+
+    /** The total time to play the animation in milliseconds */
+    public readonly duration: number;
+
+    /** Animation frames */
+    public readonly frames: AnimatedGIFFrame[];
+
+    /** Textures */
+    public readonly textures: Texture<CanvasSource>[];
+
+    /** Total number of frames in the animation */
+    public readonly totalFrames: number;
+
+    constructor(width: number, height: number, frames: AnimatedGIFFrame[])
```

Comment:
small one - wander if witdh and height could be optional? As we can get the first frame?
actually.. do we need width and height any more?

Reply:
Yeah, good catch. Removed width/height args